tokio-timer/src/timer/entry.rs

```diff
@@ -125,6 +125,7 @@ impl Entry {
     }
 
     /// Only called by `Registration`
+    #[allow(clippy::mut_from_ref)]
```

Comment:
What happens if we make the `time_mut` fn unsafe itself? That might be better anyway.

Reply:
I changed `time_mut` to unsafe fn (https://github.com/tokio-rs/tokio/pull/1335/commits/b694a3b7572cd12cecb295b70f173635e7a3ee93). But for now, it seems to be warned even if it is unsafe fn: https://github.com/rust-lang/rust-clippy/issues/4281

Reply:
Ok, allowing it here seems fine :+1: thanks for looking into it.